src/EnergyPlus/HVACControllers.cc

```diff
@@ -1130,49 +1145,49 @@ namespace HVACControllers {
 			//          earlier on the air loop.
-			if ( ! ControllerProps( ControlNum ).IsSetPointDefinedFlag ) {
-				// NOTE: For TEMPANDHUMRAT control the computed value ControllerProps(ControlNum)%SetPointValue
-				//       depends on:
-				//       - Node(SensedNode)%HumRatMax
-				//       - Node(SensedNode)%Temp
-				//       - Node(SensedNode)%HumRat
-				if ( Node( SensedNode ).HumRatMax > 0 ) {
-					// Setpoint can only be computed when the sensed node temperature is evaluated at the max
-					// actuated value for the dual humidity ratio / temperature strategy.
-					// See routine CalcSimpleController() for the sequence of operations.
-					if ( ControllerProps( ControlNum ).NextActuatedValue == RootFinders( ControlNum ).MaxPoint.X ) {
-						// Calculate the approach temperature (difference between SA dry-bulb temp and SA dew point temp)
-						ApproachTemp = Node( SensedNode ).Temp - PsyTdpFnWPb( Node( SensedNode ).HumRat, OutBaroPress );
-						// Calculate the dew point temperature at the SA humidity ratio setpoint
-						DesiredDewPoint = PsyTdpFnWPb( Node( SensedNode ).HumRatMax, OutBaroPress );
-						// Adjust the calculated dew point temperature by the approach temp
-						HumidityControlTempSetPoint = DesiredDewPoint + ApproachTemp;
-						// NOTE: The next line introduces a potential discontinuity into the residual function
-						//       which could prevent the root finder from finding the root it if were done at each
-						//       controller iteration. For this reason we perform the setpoint calculation only
-						//       once when the air loop has been evaluated with the max actuated value.
-						//       See routine CalcSimpleController() for the sequence of operations.
-						ControllerProps( ControlNum ).SetPointValue = min( Node( SensedNode ).TempSetPoint, HumidityControlTempSetPoint ); // Pure temperature setpoint | Temperature setpoint to achieve the humidity ratio setpoint
-						// Overwrite the "pure" temperature setpoint with the actual setpoint that takes into
-						// account the humidity ratio setpoint.
-						// NOTE: Check that this does not create side-effects somewhere else in the code.
-						Node( SensedNode ).TempSetPoint = ControllerProps( ControlNum ).SetPointValue;
-						// Finally indicate thate the setpoint has been computed
-						ControllerProps( ControlNum ).IsSetPointDefinedFlag = true;
-					}
-				} else {
-					// Pure temperature setpoint control strategy
-					ControllerProps( ControlNum ).SetPointValue = Node( SensedNode ).TempSetPoint;
-					// Finally indicate thate the setpoint has been computed
-					ControllerProps( ControlNum ).IsSetPointDefinedFlag = true;
-				}
+			// NOTE: For TEMPANDHUMRAT control the computed value ControllerProps(ControlNum)%SetPointValue
+			//       depends on:
+			//       - Node(SensedNode)%HumRatMax
+			//       - Node(SensedNode)%Temp
+			//       - Node(SensedNode)%HumRat
+			if ( Node( SensedNode ).HumRatMax > 0 ) {
```

Comment:
I did consider testing that, and regardless the ControllerProps.SetPointValue will have to be set. It seemed to me that the min function for ControllerProps.SetPointValue at line 1179 takes care of HumRat>HumRatMax. So my thoughts were, do I add another test in the if at line 1164 and also add an ELSE at the end to set the sensible only SP, or do I always pass through this IF with the expectation that the min function will always result in the correct answer. You can see which thought prevailed.